docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,552 @@
 
-    Visuals must be registered with a unique property index, which is used for direct access to the visual. The index is used to link a view to a visual. Registering the visual also enables additional functionality, such as connecting the visual to the window. The `RegisterVisual()` method of the [Tizen.NUI.BaseComponents.CustomView](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.CustomView.html) class stores the visual handle within the UI component.
+    ```
+    // the size of a Visual given as a percentage of the size of the parent
+    _borderVisual.RelativeSize = new RelativeVector2(0.5f, 0.5f);
+    // the parent's reference point
+    _borderVisual.Origin = Visual.AlignType.TopBegin;
+    // the Visual's reference point
+    _borderVisual.AnchorPoint = Visual.AlignType.TopBegin;
+    // shift between Origin and AnchorPoint given as a percentage of a parent size 
+    _borderVisual.RelativePosition = new RelativeVector2(0.25f, 0.25f);
+    ```
 
+4.  To add the Visual to an already existing VisualView (more details regarding VisualView available in the [next section](#visualView)) use the `AddVisual(...)` method:
     ```
-    RegisterVisual(ColorVisualPropertyIndex, _colorVisual);
+    _visualView.AddVisual("name_of_the_visual", _borderVisual);
     ```
 
-    The visuals example in this topic use property registration based on a fixed property index range. The NUI code base has been modified to perform property registration based on automatic index generation. For more information, see the [Managing Properties](customview.md#properties).
 
-5.  Set the depth index.
 
-    The depth index is the draw order for visuals within a view. Whenever a visual is added, the depth index increases automatically.
+<!---------------------------------------------------------------------------------------------------------------------------->
+<a name="visualView"></a>
+## Using VisualView
 
-    > **Note**  
-    > The recently registered visual is mostly always on top.
+A VisualView is a class that inherits after the [Tizen.NUI.BaseComponents.View](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) class.
 
+A VisualView stores handles to Visuals and maps of their properties. 
+The total number of Visuals currently added to the VisualView can be obtained via the `NumberOfVisuals` getter.
+Some useful methods for working with Visuals are:
 
-    ```
-    _colorVisual.DepthIndex = ColorVisualPropertyIndex;
-    ```
+-  To add a Visual to the VisualView one can use the `AddVisual(string, VisualMap)` method. It expects two parameters:
+   - string - a name of the Visual by which it's identified within the given VisualView
+   - [VisualMap](#visualMap) - a base class encapsulating properties of the Visual, that will be added to the view. 
+   If the Visual name passed as an argument to the function is already used in this VisualView, the Visual will be updated.
```

Comment:
I changed to:
If the Visual with a given name already exists in this VisualView it will be updated instead of added.

since this function is a part of a class the convention is to call it a method - fixed